group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -669,7 +668,12 @@ private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
                 throw new InvalidRequestException("TopicPartitions must be empty when (re-)joining.");
             }
             if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
-                throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
+                if (request.subscribedTopicRegex() == null || request.subscribedTopicRegex().isEmpty()) {
+                    throw new InvalidRequestException("SubscribedTopicNames or SubscribedTopicRegex must be set in first request.");
```

Comment:
nit: `Either SubscribedTopicNames or SubscribedTopicRegex must be set in first request.`